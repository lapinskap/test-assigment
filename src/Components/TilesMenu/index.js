import PropTypes from 'prop-types';
import { CSSTransitionGroup } from 'react-transition-group';
import React, { useContext } from 'react';
import { Col, Row } from 'reactstrap';
import { Link } from 'react-router-dom';
import PageTitle from '../../Layout/AppMain/PageTitle';
import { filterNavItemsByAlc } from '../../utils/RoleBasedSecurity/filters';
import RbsContext from '../../utils/RoleBasedSecurity/RbsContext';

export default function TilesMenu({
  title, breadcrumbs, config, pushToHistory,
}) {
  const { userInfo } = useContext(RbsContext);
  return (
    <>
      <CSSTransitionGroup
        component="div"
        transitionName="TabsAnimation"
        transitionAppear
        transitionAppearTimeout={0}
        transitionEnter={false}
        transitionLeave={false}
      >
        <PageTitle
          pushToHistory={pushToHistory}
          heading={title}
          breadcrumbs={breadcrumbs}
        />
        <div className="grid-menu grid-menu-4col">
          <Row className="no-gutters" data-t1="tilesMenu">
            {filterNavItemsByAlc(config, userInfo).map(({
              to, label, icon, tileIcon, id,
            }) => (
              <Col md="3" sm="6" className="bg-white" key={id}>
                <Link to={to} className="link-unstyled" data-t1={id}>
                  <div className="widget-chart widget-chart-hover">
                    <div className="icon-wrapper rounded-circle">
                      <div className="icon-wrapper-bg bg-primary" />
                      <i className={`${tileIcon || icon} text-primary`} />
                    </div>
                    <div className="widget-numbers-sm">{label}</div>
                  </div>
                </Link>
              </Col>
            ))}
          </Row>
        </div>
      </CSSTransitionGroup>
    </>
  );
}
TilesMenu.propTypes = {
  title: PropTypes.string.isRequired,
  pushToHistory: PropTypes.bool,
  breadcrumbs: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      link: PropTypes.string,
    }),
  ),
  config: PropTypes.arrayOf(PropTypes.shape({
    id: PropTypes.string.isRequired,
    to: PropTypes.string.isRequired,
    tileIcon: PropTypes.string,
    icon: PropTypes.string,
    label: PropTypes.string.isRequired,
  })).isRequired,
};
TilesMenu.defaultProps = {
  breadcrumbs: [],
  pushToHistory: false,
};
