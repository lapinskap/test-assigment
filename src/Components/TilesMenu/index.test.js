import React from 'react';
import renderer from 'react-test-renderer';
import TilesMenu from './index';
import '../../tests/setupTests';
import TestAppContext from '../../tests/TestAppContext';

jest.mock('react-transition-group', () => ({
  ...jest.requireActual('react-transition-group'),
  // eslint-disable-next-line react/prop-types
  CSSTransitionGroup: ({ children }) => <div>{children}</div>,
}));

describe('<TilesMenu />', () => {
  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders correct snapshot with simple props', () => {
    const config = [
      {
        id: 'test_1',
        to: '/step1/step2/test_1',
        label: 'Test 1',
        icon: 'icon_1',
      },
      {
        id: 'test_2',
        to: '/step1/step2/test_2',
        label: 'Test 2',
        icon: 'icon_2',
      },
    ];
    const tree = renderer
      .create(
        <TestAppContext router redux>
          <TilesMenu
            config={config}
            title="test title"
            breadcrumbs={[
              { title: 'Step 1', link: '/step1' },
              { title: 'Step 2', link: '/step1/step2' },
            ]}
          />
        </TestAppContext>,
      )
      .toJSON();
    expect(tree)
      .toMatchSnapshot();
  });
});
