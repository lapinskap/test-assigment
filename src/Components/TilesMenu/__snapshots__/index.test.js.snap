// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TilesMenu /> renders correct snapshot with simple props 1`] = `
<div>
  <div
    className="app-page-title app-page-title-simple"
  >
    <div
      className="page-title-wrapper"
    >
      <div
        className="page-title-heading"
      >
        <div>
          <div
            className="page-title-head center-elem"
          >
            <span
              className="d-inline-block pr-2"
            >
              <i
                className=""
              />
            </span>
            <span
              className="d-inline-block"
              data-t1="pageTitle"
            >
              test title
            </span>
          </div>
          <div
            className="page-title-subheading opacity-10"
          >
            <nav
              aria-label="breadcrumb"
              className=""
              data-t1="breadcrumbs"
            >
              <ol
                className="breadcrumb"
              >
                <li
                  className="breadcrumb-item"
                  data-t1="breadcrumbsItem"
                >
                  <a
                    href="#/"
                    onClick={[Function]}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-home fa-w-18 "
                      data-icon="home"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M280.37 148.26L96 300.11V464a16 16 0 0 0 16 16l112.06-.29a16 16 0 0 0 15.92-16V368a16 16 0 0 1 16-16h64a16 16 0 0 1 16 16v95.64a16 16 0 0 0 16 16.05L464 480a16 16 0 0 0 16-16V300L295.67 148.26a12.19 12.19 0 0 0-15.3 0zM571.6 251.47L488 182.56V44.05a12 12 0 0 0-12-12h-56a12 12 0 0 0-12 12v72.61L318.47 43a48 48 0 0 0-61 0L4.34 251.47a12 12 0 0 0-1.6 16.9l25.5 31A12 12 0 0 0 45.15 301l235.22-193.74a12.19 12.19 0 0 1 15.3 0L530.9 301a12 12 0 0 0 16.9-1.6l25.5-31a12 12 0 0 0-1.7-16.93z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </a>
                </li>
                <li
                  className="breadcrumb-item"
                  data-t1="breadcrumbsItem"
                >
                  <a
                    href="#/step1"
                    onClick={[Function]}
                  >
                    Step 1
                  </a>
                </li>
                <li
                  className="breadcrumb-item"
                  data-t1="breadcrumbsItem"
                >
                  <a
                    href="#/step1/step2"
                    onClick={[Function]}
                  >
                    Step 2
                  </a>
                </li>
                <li
                  aria-current="page"
                  className="active breadcrumb-item"
                  data-t1="breadcrumbsActiveItem"
                >
                  test title
                </li>
              </ol>
            </nav>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="grid-menu grid-menu-4col"
  >
    <div
      className="no-gutters row"
      data-t1="tilesMenu"
    >
      <div
        className="bg-white col-sm-6 col-md-3"
      >
        <a
          className="link-unstyled"
          data-t1="test_1"
          href="#/step1/step2/test_1"
          onClick={[Function]}
        >
          <div
            className="widget-chart widget-chart-hover"
          >
            <div
              className="icon-wrapper rounded-circle"
            >
              <div
                className="icon-wrapper-bg bg-primary"
              />
              <i
                className="icon_1 text-primary"
              />
            </div>
            <div
              className="widget-numbers-sm"
            >
              Test 1
            </div>
          </div>
        </a>
      </div>
      <div
        className="bg-white col-sm-6 col-md-3"
      >
        <a
          className="link-unstyled"
          data-t1="test_2"
          href="#/step1/step2/test_2"
          onClick={[Function]}
        >
          <div
            className="widget-chart widget-chart-hover"
          >
            <div
              className="icon-wrapper rounded-circle"
            >
              <div
                className="icon-wrapper-bg bg-primary"
              />
              <i
                className="icon_2 text-primary"
              />
            </div>
            <div
              className="widget-numbers-sm"
            >
              Test 2
            </div>
          </div>
        </a>
      </div>
    </div>
  </div>
</div>
`;
